	* system-config-printer.py (GUI.__init__): Don't fetch
	InstallDialog widget here...
	(NewPrinterGUI.__init__): ...fetch it here instead (bug #449860).
	(NewPrinterGUI.checkDriverExists): Fixed traceback.
Index: system-config-printer-1.0.0/system-config-printer.py
===================================================================
--- system-config-printer-1.0.0.orig/system-config-printer.py	2008-06-22 18:11:23.730825846 +0200
+++ system-config-printer-1.0.0/system-config-printer.py	2008-06-22 18:11:36.303326820 +0200
@@ -256,7 +256,6 @@ class GUI(GtkGUI, monitor.Watcher):
                         "ConnectingDialog", "lblConnecting",
                         "NewPrinterName", "entCopyName", "btnCopyOk",
                         "InfoDialog", "lblInfo",
-                        "InstallDialog", "lblInstall",
                         "AboutDialog",
                         "WaitWindow", "lblWait",
                         )
@@ -2354,7 +2353,8 @@ class NewPrinterGUI(GtkGUI):
                         "rbtnNPDownloadLicenseYes",
                         "rbtnNPDownloadLicenseNo",
                         "NewPrinterName", "entCopyName", "btnCopyOk",
-                        "InfoDialog", "lblInfo")
+                        "InfoDialog", "lblInfo",
+                        "InstallDialog", "lblInstall")
         # share with mainapp
         self.WaitWindow = mainapp.WaitWindow
         self.lblWait = mainapp.lblWait
@@ -4701,7 +4701,7 @@ class NewPrinterGUI(GtkGUI):
                                 (name, pkg))
                 dialog = self.InstallDialog
                 self.lblInstall.set_markup(install_text)
-                dialog.set_transient_for (self.MainWindow)
+                dialog.set_transient_for (self.mainapp.MainWindow)
                 response = dialog.run ()
                 dialog.hide ()
                 if response == gtk.RESPONSE_OK:
@@ -4712,7 +4712,7 @@ class NewPrinterGUI(GtkGUI):
                                      "but it is not currently installed.  "
                                      "Please install it before using this "
                                      "printer.") % (name, (exes + pkgs)[0]),
-                                   self.MainWindow)
+                                   self.mainapp.MainWindow)
 
 
 def main(start_printer = None, change_ppd = False):
